frontend/app.js

```diff
@@ -0,0 +1,510 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
```

Comment:
Also, as you've mentioned, this file is pretty huge. The general rule is one component per file. The convention for file names is that they should be named whatever their default export is.

So, in the case of a file that exports a component called `TaskList`, you'd have `TaskList.js`. For a file that exports a function called `calculate`, you'd have `calculate.js`.